plugin/rename_file.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+from .core.open import open_file_uri
+from .core.protocol import Notification, RenameFilesParams, Request, WorkspaceEdit
+from .core.registry import LspWindowCommand
+from .core.types import match_file_operation_filters
+from pathlib import Path
+from urllib.parse import urljoin
+import os
+import sublime
+import sublime_plugin
+
+
+class LspRenameFromSidebarOverride(LspWindowCommand):
+    def is_enabled(self):
+        return True
+
+    def run(self, paths: list[str] | None = None) -> None:
+        old_path = paths[0] if paths else None
+        if old_path:
+            self.window.run_command('lsp_rename_path', {
+                "old_path": old_path
+            })
+
+
+class RenamePathInputHandler(sublime_plugin.TextInputHandler):
+    def __init__(self, path: str) -> None:
+        self.path = path
+
+    def name(self) -> str:
+        return "new_name"
+
+    def placeholder(self) -> str:
+        return self.path
+
+    def initial_text(self) -> str:
+        return self.placeholder()
+
+    def initial_selection(self) -> list[tuple[int, int]]:
+        name, _ext = os.path.splitext(self.path)
+        return [(0, len(name))]
+
+    def validate(self, path: str) -> bool:
+        return len(path) > 0
+
+
+class LspRenamePathCommand(LspWindowCommand):
+    capability = 'workspace.fileOperations.willRename'
+
+    def is_enabled(self):
+        return True
+
+    def want_event(self) -> bool:
+        return False
+
+    def input(self, args: dict) -> sublime_plugin.TextInputHandler | None:
+        if "new_name" in args:
+            return None
+        view = self.window.active_view()
+        old_path = view.file_name() if view else None
+        return RenamePathInputHandler(Path(old_path or "").name)
+
+    def run(
+        self,
+        new_name: str,  # new_name can be: FILE_NAME.xy OR ./FILE_NAME.xy OR ../../FILE_NAME.xy
```

Comment:
Perhaps this comment belong better where we normalize it into `new_path`?
And then signature can be compressed into one line.